src/editors/playgroundController.ts

```diff
@@ -0,0 +1,125 @@
+import * as vscode from 'vscode';
+
+import ConnectionController from '../connectionController';
+import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
+import { CompassServiceProvider } from '@mongosh/service-provider-server';
+import ActiveDBCodeLensProvider from './activeDBCodeLensProvider';
+import formatOutput from '../utils/formatOutput';
+import { createLogger } from '../logging';
+import { OutputChannel } from 'vscode';
+
+const log = createLogger('editors controller');
+
+/**
+ * This controller manages playground.
+ */
+export default class PlaygroundController {
+  _context?: vscode.ExtensionContext;
+  _connectionController?: ConnectionController;
+  _runtime?: ElectronRuntime;
+  _activeDB?: any;
+  _activeDBCodeLensProvider?: ActiveDBCodeLensProvider;
+  _outputChannel: OutputChannel;
+
+  constructor(context: vscode.ExtensionContext, connectionController: ConnectionController) {
+    this._context = context;
+    this._connectionController = connectionController;
+    this._outputChannel = vscode.window.createOutputChannel(
+      'Playground output'
+    );
+  }
+
+  createPlayground(): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      vscode.workspace
+        .openTextDocument({
+          language: 'mongodb',
+          content: '// The MongoDB playground'
```

Comment:
I wish we could also provide a starting name for the text document here so that users know to save it as a `.mongodb` extension, but I don't think we can... I think we've talked about this with Max - later on it'll be nice to have a good template for the user to start with so they have an understanding of the capability of playgrounds. Seeing if we have a ticket for that

Reply:
For new documents, VSCode shows an `untitled` filename without an extension. Then as you type it replaces it with the first line form the active document. And only after saving it adds an extension to the custom title. There is a way how to hack the title, but as I read the way how it works was intentionally implemented by VSCode. And the extension is being added only after saving anyway. This is actually the reason why I check `editorLangId` here and not `resourceLangId`.

https://github.com/mongodb-js/vscode/blob/VSCODE-47-integration-with-mongosh/package.json#L377